src/nemo_evaluator/observability/model_traffic.py

```
@@ -376,6 +382,110 @@ def _is_success(record: dict[str, Any]) -> bool:
+    def _load(self) -> None:
+        if self._loaded_records is None:
+            self._loaded_records = list(self._iter_file())
+
+    def _iter_file(self) -> Iterator[dict[str, Any]]:
+        path = self._path
+        if path is None:
+            return
+        try:
+            with path.open(encoding="utf-8") as f:
+                for line in f:
+                    line = line.strip()
```

Ruff (0.15.21)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)